sktime/forecasting/trend/_stl_forecaster.py

```diff
@@ -180,6 +179,21 @@ def __init__(
         self.forecaster_resid = forecaster_resid
         super().__init__()
 
+        for forecaster in (
```

Comment:
A short disclaimer: I am a big fan of lambda expressions. 

I think you could replace your code block with: 

```python
ignore_exogenous = all(map(lambda f: f is not None and f.get_tag("ignores-exogeneous-X"),
                                   (forecaster.forecaster_trend, forecaster.forecaster_seasonal, forecaster.forecaster_resid)))
self.set_tags(**{"ignores-exogeneous-X": ignore_exogenous})
```
This lambda expression should result in the same code.

However, if you think this expression is hard to understand, feel free to ignore this comment :)

Reply:
I think the linter complains about lambda expressions, it wants def statements instead...

I also like lambdas but have repeatedly found the linter to complain.
But: if it's nested, it's harder to read. If it's separate and you substitute, the linter does not like it.

Reply:
I've similar experience with linters, and may or may not be because of that, somehow have convinced myself that longer lambda's are usually difficult to follow when I come back to that code later. Not making this change now, but thanks for the suggestion.